test/cluster/object_store/test_backup.py

```diff
@@ -359,6 +360,148 @@ async def test_simple_backup_and_restore(manager: ManagerClient, s3_server):
+    for server in servers:
+        await manager.api.flush_keyspace(server.ip_addr, keyspace)
+
+    # Take snapshot for keyspace
+    snapshot_name = unique_name('backup_')
+    print(f"Taking snapshot '{snapshot_name}' for keyspace '{keyspace}'...")
+    for server in servers:
+        await manager.api.take_snapshot(server.ip_addr, keyspace, snapshot_name)
+
+    # Collect snapshot files from each server
+    async def get_snapshot_files(server, snapshot_name):
+        workdir = await manager.server_get_workdir(server.server_id)
+        data_path = os.path.join(workdir, 'data', keyspace)
+        cf_dirs = os.listdir(data_path)
+        if not cf_dirs:
+            raise RuntimeError(f"No column family directories found in {data_path}")
+        # Assumes that there is only one column family directory under the keyspace.
+        cf_dir = cf_dirs[0]
+        snapshot_path = os.path.join(data_path, cf_dir, 'snapshots', snapshot_name)
+        return [
+            f.name for f in os.scandir(snapshot_path)
+            if f.is_file() and f.name.endswith('TOC.txt')
+        ]
+
+    sstables = {}
+    for server in servers:
+        snapshot_files = await get_snapshot_files(server, snapshot_name)
+        sstables[server.server_id] = snapshot_files
+
+    # Backup the keyspace on each server to S3
+    prefix = f"{table}/{snapshot_name}"
+    print(f"Backing up keyspace using prefix '{prefix}' on all servers...")
+    for server in servers:
+        backup_tid = await manager.api.backup(
+            server.ip_addr,
+            keyspace,
+            table,
+            snapshot_name,
+            s3_server.address,
+            s3_server.bucket_name,
+            prefix
+        )
+        backup_status = await manager.api.wait_task(server.ip_addr, backup_tid)
+        assert backup_status is not None and backup_status.get('state') == 'done', \
+            f"Backup task failed on server {server.server_id}"
+
+    # Truncate data and start restore
+    print("Dropping table data...")
+    cql.execute(f"TRUNCATE TABLE {keyspace}.{table};")
+    print("Initiating restore operations...")
+
+    restore_task_ids = {}
+    for server in servers:
+        restore_tid = await manager.api.restore(
+            server.ip_addr,
+            keyspace,
+            table,
+            s3_server.address,
+            s3_server.bucket_name,
+            prefix,
+            sstables[server.server_id]
+        )
+        restore_task_ids[server.server_id] = restore_tid
+
+    await asyncio.sleep(0.1)
+
+    print("Aborting restore tasks...")
+    for server in servers:
+        await manager.api.abort_task(server.ip_addr, restore_task_ids[server.server_id])
+
+    # Check final status of restore tasks
+    for server in servers:
+        final_status = await manager.api.wait_task(server.ip_addr, restore_task_ids[server.server_id])
+        print(f"Restore task status on server {server.server_id}: {final_status}")
```

Comment:
Shouldn't the test have a check on `final_status` too?

Reply:
added assert, just in case